packages/iris-grid/src/IrisGridTableModel.js

```diff
@@ -458,35 +460,34 @@ class IrisGridTableModel extends IrisGridModel {
     return '';
   }
 
-  getColumnsWithLayoutHints = memoize((columns, hints) => {
-    if (hints) {
+  getColumnsWithLayoutHints = memoize((columns, hints, userFrozenColumns) => {
```

Comment:
I see why I was getting so confused... this is still called `userFrozenColumns` here, but you're passing in `this.frozenColumns`, which is the userFrozenColumns OR the layoutHints frozen columns.
You could just rename this var to `frozenColumnsParam` and be done with it... I think what would be even clearer though would be changing this signature, e.g.:
```
getColumnsWithLayoutHints = memoize((columnMap, frontColumns, backColumns, frozenColumns) => {
  ...
}
```

Then `get columns()` changes to:
```
return this.getColumnsWithLayoutHints(this.columnMap, this.frontColumns, this.backColumns, this.frozenColumns)
```
Then you add memoized functions for columnMap, frontColumns, backColumns similar to frozenColumns, eg.
```
getMemoizedColumnMap = memoize((tableColumns) => {
  const columnMap = new Map();
  columns.forEach(col => columnMap.set(col.name, col));
  return columnMap;
})

get columnMap() {
  return this.getColumnMap(this.table.columns);
}

getMemoizedFrontColumns = memoize((layoutHintsFrontColumns) => {
   return layoutHintsFrontColumns ?? [];
});

get frontColumns() {
  return this.getMemoizedFrontColumns(this.layoutHints?.frontColumns);
}

getMemoizedBackColumns = memoize((layoutHintsBackColumns) => {
   return layoutHintsBackColumns ?? [];
});

get backColumns() {
  return this.getMemoizedFrontColumns(this.layoutHints?.backColumns);
}

getMemoizedFrozenColumns = memoize((layoutHintsFrozenColumns, userFrozenColumns) => {
   return userFrozenColumns ?? layoutHintsFrozenColumns ?? [];
});

get frozenColumns() {
  return this.getMemoizedFrozenColumns(this.layoutHints?.frozenColumns, this.userFrozenColumns);
}
```